File: lost/backend/app/models/post.py
import uuid
from datetime import datetime, timezone


class Post:
    """Represents a lost-or-found item post."""

    def __init__(
        self,
        title: str,
        description: str,
        category: str,
        location: str,
        post_type: str,           # "lost" | "found"
        image_url: str | None = None,
        latitude: float | None = None,
        longitude: float | None = None,
    ):
        self.id = str(uuid.uuid4())
        self.title = title
        self.description = description
        self.category = category
        self.location = location
        self.type = post_type
        self.image_url = image_url
        self.latitude = latitude
        self.longitude = longitude
        self.created_at = datetime.now(timezone.utc).isoformat()

    def to_dict(self) -> dict:
        return {
            'id': self.id,
            'title': self.title,
            'description': self.description,
            'category': self.category,
            'location': self.location,
            'type': self.type,
            'image_url': self.image_url,
            'latitude': self.latitude,
            'longitude': self.longitude,
            'created_at': self.created_at,
        }

File: lost/lib/domain/entities/post.dart
/// Post Entity - Domain Layer
class Post {
  final String id;
  final String userId;
  final String title;
  final String description;
  final String category;
  final String postType; // 'lost' or 'found'
  final String imageUrl;
  final String country;
  final String? state;
  final String? city;
  final double? latitude;
  final double? longitude;
  final String? location;
  final DateTime createdAt;
  final DateTime? updatedAt;

  const Post({
    required this.id,
    required this.userId,
    required this.title,
    required this.description,
    required this.category,
    required this.postType,
    required this.imageUrl,
    required this.country,
    this.state,
    this.city,
    this.latitude,
    this.longitude,
    this.location,
    required this.createdAt,
    this.updatedAt,
  });

  Post copyWith({
    String? id,
    String? userId,
    String? title,
    String? description,
    String? category,
    String? postType,
    String? imageUrl,
    String? country,
    String? state,
    String? city,
    double? latitude,
    double? longitude,
    String? location,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Post(
      id: id ?? this.id,
      userId: userId ?? this.userId,
      title: title ?? this.title,
      description: description ?? this.description,
      category: category ?? this.category,
      postType: postType ?? this.postType,
      imageUrl: imageUrl ?? this.imageUrl,
      country: country ?? this.country,
      state: state ?? this.state,
      city: city ?? this.city,
      latitude: latitude ?? this.latitude,
      longitude: longitude ?? this.longitude,
      location: location ?? this.location,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}
